docs/standard/native-interop/com-wrappers-source-generation.md

```diff
@@ -0,0 +1,166 @@
+IFoo foo = GetManagedIFoo();
+nint ptr = Marshal.GetIUnknownForObject(foo);
+GivePointerToComInterface(nint ptr);
+```
+
+The COM wrappers API enables interacting with COM in C# without using the the built-in COM system, but requires lots of boilerplate and hand written unsafe code. The COM interface generator automates that process and makes COM wrappers as easy as built-in COM.
+
+## Basic Usage
+
+To use the COM interface generator, add a <xref:System.Runtime.InteropServices.GeneratedComInterfaceAttribute> and <xref:System.Runtime.InteropServices.GuidAttribute> on the interface definition that you want to import from COM or expose to COM. The type will also need to be `partial` and have `internal` or `public` visibility for the generated code to be able to access it.
+
+```csharp
+[GeneratedComInterface]
+[Guid("3faca0d2-e7f1-4e9c-82a6-404fd6e0aab8")]
+internal partial IFoo
+{
+    void Method(int i);
+}
+```
+Then, if you want to expose a class that implements an interface to COM, add the <xref:System.Runtime.InteropServices.GeneratedComClassAttribute> to the implementing class. This class will also need to be `partial` and either `internal` or `public`.
+
+```csharp
+[GeneratedComClass]
+internal partial class Foo : IFoo
+{
+    public void Method(int i)
+    {
+        // Do things
+    }
+}
+```
+
+At compile time, the generator will create an implementation of the COM wrappers API, and you can use the <xref:System.Runtime.InteropServices.StrategyBasedComWrappers> type or a custom derived type to consume or expose the COM interface.
+
+```csharp
+[LibraryImport("MyComObjectProvider.dll")]
+static nint GetPointerToComInterface();
+
+[LibraryImport("MyComObjectProvider.dll")]
+static void GivePointerToComInterface(nint comObject);
+
+// Use the system to create a Runtime Callable Wrapper to use in managed code
+ComWrappers cw = new StrategyBasedComWrappers();
+nint ptr = GetPointerToComInterface();
+IFoo foo = cw.GetOrCreateObjectForComInterface(ptr);
+foo.Method(0);
+...
+// Use the system to create a COM Callable Wrapper to pass to unmanaged code
+ComWrappers cw = new StrategyBasedComWrappers();
+Foo foo = new Foo();
+nint ptr = cw.GetOrCreateComInterfaceForObject(foo);
+GivePointerToComInterface(nint ptr);
+```
+
+## Customizing Marshalling
+
+The COM interface generator respects the <xref:System.Runtime.InteropServices.Marshalling.MarshalAsAttribute> and <xref:System.Runtime.InteropServices.Marshalling.MarshalUsingAttribute> attributes to customize marshalling of parameters. For more information, see [Customize Parameter Marshalling](./customize-parameter-marshalling.md) If an interface uses `string`s, GeneratedComInterfaceAttribute has StringMarshalling and StringMarshallingCustomType properties that will apply to all parameters and return types of `string` in the interface if they don't have other marshalling attributes.
+
+## Implicit HRESULTs and PreserveSig
```

Comment:
I think this explanation is actually helpful for both built-in and source-generated COM. Would it make sense to pull this out into a different page?

Reply:
The closest existing doc I saw is the examples in the `PreserveSig` API doc (which is explicitly calling out tlbexp/tlbimp): https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.preservesigattribute#examples